app/src/main/java/com/valhalla/thor/data/launcher/FreezerShortcutManager.kt

```
@@ -0,0 +1,263 @@
+    // Rebuild + push the current-state icon for a package's pinned shortcut (no-op if absent).
+    private fun updateShortcutIcon(packageName: String) {
+        val label = appLabel(packageName) ?: return
+        ShortcutManagerCompat.updateShortcuts(context, listOf(buildAppShortcut(packageName, label)))
+    }
+
+    // "Frozen" == the app is currently disabled. MATCH_DISABLED_COMPONENTS so we can still read it.
+    private fun isFrozen(packageName: String): Boolean = try {
+        !context.packageManager
+            .getApplicationInfo(packageName, PackageManager.MATCH_DISABLED_COMPONENTS)
+            .enabled
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)